src/trg-main-window.c

```diff
@@ -1721,8 +1721,12 @@ trg_main_window_conn_changed(TrgMainWindow * win, gboolean connected)
 
         trg_torrent_model_remove_all(priv->torrentModel);
 
-        g_source_remove(priv->timerId);
-        g_source_remove(priv->sessionTimerId);
+        if (priv->timerId > 0)
```

Comment:
Use `g_clear_handle_id (&priv->timerId);` for all of these.

It handles `0`, and it sets to `0` after to avoid dangling invalid ids.

Reply:
This does mean depending on glib 2.56, you can bump that in `configure.ac` and match it in the meson branch.